### PR TITLE
Fix autoloading

### PR DESCRIPTION
**Pull request by W0rma**
The directory structure in the folder "tests" is not compatible with [PSR-0](https://www.php-fig.org/psr/psr-0/). 

Therefore ["classmap"](https://getcomposer.org/doc/04-schema.md#classmap) should be used to ensure proper autoloading in composer. Since https://github.com/composer/composer/pull/8652 which is part of the latest version 1.10 composer shows a warning.

### CODE REVIEW

**Comment by dereuromark** [comment]
We have the same issue.
Because oif new composer we get now those warnings as well

Might also be worth properly separating autoload and -dev, as tests should only be exposed through the latter, and not contaminate production map.
On top, it might be valuable to move to psr4 namespacing now.

**Comment by W0rma** [comment]
> Might also be worth properly separating autoload and -dev, as tests should only be exposed through the latter, and not contaminate production map.
> On top, it might be valuable to move to psr4 namespacing now.

Good idea. I have updated my PR. PSR-4 is now used for `src/` and `tests/`. Tests have been moved to `autoload-dev`.

**Review by boesing — APPROVED** [review]
LGTM. Having `autoload-dev` is also a good extension to this project as there should be no need for test assets in any implementing project.

Having a [.gitattributes](https://github.com/mezzio/mezzio/blob/master/.gitattributes) file would help to exclude some files (e.g. Testassets) from the packagist archive aswell.

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ php:
   - '7.1'
   - '7.2'
   - '7.3'
+  - '7.4'
   - 'nightly'
```
Comment (boesing) [inline .travis.yml:7]: I would drop this to make tests pass. There is actually no need to test against PHP 8.0 imho

**Comment by boesing** [comment]
@stephenfrank Would you merge this if tests pass?
Are you fine with removing nightly PHP 8 from tests?
Or probably moving nightly to `allow failures` instead?

File: .travis.yml (hunk 1 above)
Comment (W0rma) [reply .travis.yml:7]: Good point. I moved nightly to `allow failures`

**Comment by martinssipenko** [comment]
Any updates on this?

**Review by boesing — APPROVED** [review]

**Review by boesing — APPROVED** [review]

**Comment by fr05t1k** [comment]
https://github.com/FlowCommunications/JSONPath/issues/52

**Review by Jeckerson — APPROVED** [review]

**Comment by Jeckerson** [comment]
Got same error message today.

**Review by martinssipenko — APPROVED** [review]

**Comment by dereuromark** [comment]
@W0rma Whats the hold up? The PR was fine months ago. Many projects are facing this issue right now indirectly (by using a library that uses this one).

**Comment by simensen** [comment]
Is there anything we can do to help get this merged and a new tag cut? :) Would love to help if I can!

**Review by andrew-demb — APPROVED** [review]

**Review by martinssipenko — APPROVED** [review]

**Comment by martinssipenko** [comment]
This project is now marked as abandoned and suggests using [`softcreatr/jsonpath`](https://github.com/SoftCreatR/JSONPath) as a replacement. If you think this issue is still relevant please open a new ticket under the new project.

https://github.com/SoftCreatR/JSONPath

**Pull request closed without merging**